e2e/config/azure.go

```diff
@@ -491,3 +492,58 @@ func ensureProvisioningState(version *armcompute.GalleryImageVersion) error {
 	}
 	return nil
 }
+
+func (a *AzureClient) CreateVMSSWithRetry(ctx context.Context, t *testing.T, resourceGroupName string, vmssName string, parameters armcompute.VirtualMachineScaleSet) (*armcompute.VirtualMachineScaleSet, error) {
+	t.Logf("creating VMSS %s in resource group %s", vmssName, resourceGroupName)
+	delay := 5 * time.Second
```

Comment:
Why 5 seconds? If we hit a quota issue, then 5 seconds likely isn't going to be enough. Actually, why are we retrying at all? It'd be good to have a comment about which types of failures the retry is trying to resolve - that'll help the next developer.

Reply:
it's a few lines below. Some ARM-specific issues on new GPUs with low availability.

I suspect it's something to do with the VM being recycled. And I retry tests faster than it's cleaned